packages/main/src/SliderBase.ts

```diff
@@ -795,12 +795,19 @@ abstract class SliderBase extends UI5Element {
 		return this.editableTooltip ? "ui5-slider-InputDesc" : undefined;
 	}
 
-	get ariaLabelText() {
-		return getAssociatedLabelForTexts(this);
-	}
+	get _ariaLabel() {
+		const associatedLabelText = getAssociatedLabelForTexts(this);
+		const hasAccessibleName = !!this.accessibleName;
+
+		let labelText = hasAccessibleName
+			? `${this.accessibleName} ${this._ariaLabelledByText}`
+			: this._ariaLabelledByText;
+
+		if (associatedLabelText && !hasAccessibleName) {
```

Comment:
You can check first for !hasAccessibleName and then for associatedLabelText to be more clear that the accessibleName is more important and the "for" attribute will be considered only in case there is no accessibleName set